add_me_to_your_PATH/Dockerfile.template

```diff
@@ -93,7 +93,7 @@ FROM sysimage-image as precompile-image
 ARG ADD_UTILS="true"
 RUN --mount=type=secret,id=github_token \
     if [ "$ADD_UTILS" = "true" ]; then \
-        julia -e 'using Pkg; Pkg.add("Revise"); Pkg.add("PProf"); Pkg.instantiate()'; \
+        julia -e 'using Pkg; Pkg.add("Revise"; preserve=Pkg.PRESERVE_ALL); Pkg.add("PProf"; preserve=Pkg.PRESERVE_ALL); Pkg.instantiate()'; \
```

Comment:
what's the logic of doing the `Pkg.add` before `Pkg.instantiate`? I would think you'd first install all the versions from the manifest, and _then_ add the additional packages (while maintaining the manifest versions). Is it possible here that the `Pkg.add` will install the latest versions of the packages and then `Pkg.instantiate `will lock them to the manifest afterwards?

Reply:
Looking at the `Pkg.jl` docs, looks like `instantiate` does not `resolve` anything if a Manifest already exists, which it should after calling `Pkg.add`.

My guess is that that `Pkg.instantiate` call is a no-op here, but I haven't checked to be sure.